### PR TITLE
Fix: fixed connect API for 3 providers (Github, Google and Facebook)

### LINKED ISSUE #6570: Cannot get any authentication provider to work: redirect_uri_mismatch
**Describe the bug**
I followed the documentation, I tried using the react-login-front-end-app, no matter how hard I try, I always run into the same redirect_uri_mismatch error.

**Steps to reproduce the behavior**
1. Select any provider [in your strapi admin panel](http://localhost:1337/admin/plugins/users-permissions/providers), let's go with GitHub, since it seems to be the simplest and also has a [video tutorial from Strapi](https://www.youtube.com/watch?v=s_vQqQ4tpHs) (tried following it step by step).
2. Fill in the required fields as [described in the docs](https://strapi.io/documentation/v3.x/plugins/users-permissions.html#providers); also follow the other steps described in the docs
3. Go to http://localhost:1337/connect/github or click on the Github button inside the React login frontend boilerplate app

**Expected behavior**
Successful authentication

**Screenshots**
See in **Additional context**

**System**
- Node.js version: 12.16.1
- NPM version: 6.14.5
- Strapi version: 3.0.1
- Database: Postgres 12.3
- Operating system: Windows 10 Pro x64

**Additional context**
Seems to be a problem with how strapi is sending the redirect_uri to the Oauth providers. Shouldn't it be something like https://760bbe10e952.ngrok.io/connect/github/callback ?
image
image

I also noticed that, as opposed to the tutorial video, redirect URL field in the provider config doesn't get automatically populated, could it be a frontend issue?
image

If I try any other provider (tried facebook & google as well), I also get an URL related error.

### PR DESCRIPTION
**Pull request by reubenkcoutinho**
## Description of what you did:
- Modified user-permission plugin to fix connect API for github, facebook and google providers
- Added documentation of how to possible override connect values 

## related issue fix
#6570 

NOTE: this might be considered a WIP as I haven't had the time to complete the remaining providers

### CODE REVIEW

File: packages/strapi-plugin-users-permissions/README.md [hunk 1]
```diff
@@ -1 +1,23 @@
-# Strapi plugin
\ No newline at end of file
+# Strapi plugin
+
+After you need to set the provider values in the admin panel, you can override the srapi provided value using the following method.
```
Comment (votemike) [inline packages/strapi-plugin-users-permissions/README.md:3]:
```suggestion
After you need to set the provider values in the admin panel, you can override the strapi provided value using the following method.
```

**Comment by petersg83** [comment]
Hi :)
Thanks for your contribution. I don't understand what is broken. I tried the authentication flow with github and google and it worked well. Here is an example of what I did for google : https://youtu.be/8yAiBJvr62Q
Can you explain what you fell is not working please ?

I think the problem is the doc that is not clear enough. I'll take time to update it :)

**Comment by reubenkcoutinho** [comment]
Hi @petersg83 , this fix was for this issue [(6570)](https://github.com/strapi/strapi/issues/6570). This was on Strapi 3.0.0.

I haven't verified if it works now using an updated Strapi version. Could you verify this?

If it is fixed, great. I'll decline the pull request.

**Comment by petersg83** [comment]
Hi, thanks for the details! I just tried with 3.0.0 and it also works. I'll update the doc and close the issue and the PR. Don't hesite to reopen one if needed!
Have a great day :)

**Comment by petersg83** [comment]
Will be fixed by this PR : #6723

File: packages/strapi-plugin-users-permissions/controllers/Auth.js [hunk 2]
```diff
@@ -256,7 +256,11 @@ module.exports = {
     }
     // Ability to pass OAuth callback dynamically
     grantConfig[provider].callback = _.get(ctx, 'query.callback') || grantConfig[provider].callback;
-    grantConfig[provider].redirect_uri = `${strapi.config.server.url}/connect/${provider}/callback`;
+    grantConfig[provider].redirect_uri =
+      (strapi.config.provider &&
+        strapi.config.provider[provider] &&
+        strapi.config.provider[provider].redirect_url) ||
```
Comment (votemike) [inline packages/strapi-plugin-users-permissions/controllers/Auth.js:262]: Can you use `_.get()` here?

File: packages/strapi-plugin-users-permissions/services/Providers.js [hunk 3]
```diff
@@ -121,6 +121,25 @@ const getProfile = async (provider, query, callback) => {
     })
     .get();
 
+  const client_id =
+    (strapi.config.provider &&
+      strapi.config.provider[provider] &&
+      strapi.config.provider[provider].client_id) ||
```
Comment (votemike) [inline packages/strapi-plugin-users-permissions/services/Providers.js:127]: Can you use `_.get()` here?

Comment (reubenkcoutinho) [reply packages/strapi-plugin-users-permissions/services/Providers.js:127]: Thanks @votemike but @petersg83 closed the pull request

**Pull request closed without merging**